packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -579,9 +579,9 @@ export default defineComponent({
     },
 
     getAvailableRoleOptions(link) {
-      if (this.incomingParentShare.value && this.canCreatePublicLinks) {
+      if (this.incomingParentShare && this.canCreatePublicLinks) {
         return LinkShareRoles.filterByBitmask(
-          this.incomingParentShare.value.permissions,
+          this.incomingParentShare.permissions,
```

Comment:
Same here I think, `.value` is needed.

Reply:
I'm pretty sure it's not.

It was with `@vue/composition-api` but now it's not anymore. Feel free to test in browser :)

Reply:
But it's because of `inject: ['incomingParentShare']`, that's not related to the composition API, or? If we need to remove it, then we need to apply this change to all occurrences of `incomingParentShare`, not just here.

Reply:
There aren't many occurences and I couldn't find that this breaks anything, maybe you can check once again.

Reply:
fwiw I think half of the acceptance tests were broken without this change, with it all acceptance tests suddenly passed

Reply:
Seems to work, you're right. So did it change in Vue 2.7? I remember adding `.value` a few weeks ago either here or in `RoleDropdown.vue` because it was missing and broken... that's why I was confused about it.

Reply:
Ahhh maybe related to `provide('incomingParentShare', incomingParentShare)` which is new? Because now it is provided via `setup()`.

Reply:
Yes, something changed with the built in composition api in Vue 2.7.
I'm not sure it depends on the syntax used, tho. I think it worked with the old syntax as well after this change, but I kept the new syntax because I had tried that earlier and thought it was worth keeping it...

Reply:
Absolutely, it's way nicer this way 👍